lib/matplotlib/axes/_base.py

```diff
@@ -167,23 +165,67 @@ def set_prop_cycle(self, *args, **kwargs):
     def __call__(self, *args, **kwargs):
         if self.axes.xaxis is not None and self.axes.yaxis is not None:
             xunits = kwargs.pop('xunits', self.axes.xaxis.units)
-
             if self.axes.name == 'polar':
                 xunits = kwargs.pop('thetaunits', xunits)
+            if xunits != self.axes.xaxis.units:
```

Comment:
?? whats this for?  Seems orthogonal to the PR, and I don't think we've agreed on the units handling.  Not saying its wrong, but...

Reply:
doesn't really matter (was just shuffling a few lines around to make things more compact) and reverted it